src/main/java/com/ait/lienzo/client/core/types/BoundingPoints.java

```diff
@@ -35,13 +37,40 @@ public BoundingPoints(final BoundingBox bbox)
 
         double h = bbox.getHeight();
 
-        m_array.push(new Point2D(x + 0, y + 0));
+        double x0;
+        double y0;
+        double x1;
+        double y1;
+
+        if (bbox.getMaxX() > bbox.getMinX())
+        {
+            x0 = x + 0;
```

Comment:
The BoundingBox does not understand about directions, IMO there is no need for, the box and the points resulting are two different things, because as you say, direction matters, so the points for the box depend on it....